skimage/measure/_regionprops.py

```diff
@@ -729,12 +728,24 @@ def _props_to_dict(regions, properties=('label', 'bbox'), separator='-'):
             else:
                 shape = (len(rp),)
 
+            # precompute property column names and locations
+            modified_props = []
+            locs = []
             for ind in np.ndindex(shape):
-                for k in range(n):
-                    loc = ind if len(ind) > 1 else ind[0]
-                    column_buffer[k] = regions[k][prop][loc]
-                modified_prop = separator.join(map(str, (prop,) + ind))
-                out[modified_prop] = np.copy(column_buffer)
+                modified_props.append(separator.join(map(str, (prop,) + ind)))
+                locs.append(ind if len(ind) > 1 else ind[0])
+
+            # fill temporary column data_array
+            n_columns = len(locs)
+            column_data = np.empty((n, n_columns), dtype=dtype)
+            for k in range(n):
+                rp = regions[k][prop]
+                for i, loc in enumerate(locs):
+                    column_data[k, i] = rp[loc]
+
+            # add the columns to the output dictionary
+            for i, modified_prop in enumerate(modified_props):
+                out[modified_prop] = column_data[:, i]
```

Comment:
This reordering of the code here is the primary source of the performance improvement seen in the benchmarks.